### PR TITLE
use  Java 17, fix compile warnings [AJ-382]

### PR DESCRIPTION
**Pull request by davidangb**
1. Update Orchestration to use and compile to Java 17. See companion PR broadinstitute/firecloud-develop#2879 for the local dev environment.
2. Removes compile-time warnings about ["in" vs slash syntax ](https://www.scala-sbt.org/1.x/docs/Migrating-from-sbt-013x.html#slash)in sbt project build files
3. Removes, from the automation subproject, compile-time warnings left over from the Scala 2.13 upgrade at #930
4. Upgrades sbt from 1.4.9 to 1.6.2

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/OrchestrationApiSpec.scala [hunk 1]
```diff
@@ -69,7 +69,7 @@ class OrchestrationApiSpec
         verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", true), NihDatasetPermission("TARGET", true)))
         Orchestration.NIH.addUserInNIH(OrchConfig.Users.genericJsonWebTokenKey)
         verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", false), NihDatasetPermission("TARGET", false)))
-      } finally resetNihLinkToInactive
+      } finally resetNihLinkToInactive()
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/OrchestrationApiSpec.scala:72]: these are Scala 2.13 fixes that we missed in #930

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/StorageApiSpec.scala [hunk 2]
```diff
@@ -30,7 +29,6 @@ import scala.language.implicitConversions
 
 class StorageApiSpec extends AnyFreeSpec with StorageApiSpecSupport with Matchers with LazyLogging {
   implicit val system: ActorSystem = ActorSystem()
-  implicit val materializer: ActorMaterializer = ActorMaterializer()
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/StorageApiSpec.scala:33]: a Scala 2.13 fix that we missed in #930 - ActorMaterializer is now automatically found from the ActorSystem, we don't need to create one ourselves

File: build.sbt [hunk 3]
```diff
@@ -14,4 +14,4 @@ Revolver.enableDebugging(port = 5051, suspend = false)
 // itself rather than sbt, but they are not passed by default to the application, which is a forked
 // process. This passes them through to the "re-start" command, which is probably what a developer
 // would normally expect.
-javaOptions in reStart ++= sys.env("JAVA_OPTS").split(" ").toSeq
+reStart / javaOptions ++= sys.env("JAVA_OPTS").split(" ").toSeq
```
Comment (davidangb) [inline build.sbt:17]: see https://www.scala-sbt.org/1.x/docs/Migrating-from-sbt-013x.html#slash

File: project/build.properties [hunk 4]
```diff
@@ -1 +1 @@
-sbt.version=1.4.9
+sbt.version=1.6.2
```
Comment (davidangb) [inline project/build.properties:1]: #930 upgraded various docker base images to `hseeberger/scala-sbt:eclipse-temurin-11.0.14.1_1.6.2_2.13.8` - which uses sbt 1.6.2. However, this build.properties still specified 1.4.9. Let's fix that.

File: Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM us.gcr.io/broad-dsp-gcr-public/base/jre:11-debian
+FROM us.gcr.io/broad-dsp-gcr-public/base/jre:17-debian
```
Comment (sagehen03) [inline Dockerfile:1]: I know this predates this change, but why do we use the debian image here and temurin elsewhere?

**Review by sagehen03 — APPROVED** [review]
looks good as far as I can tell.  Trusting the compiler to save us in pre-prod environments if there are any issues.

File: Dockerfile (hunk 5 above)
Comment (davidangb) [reply Dockerfile:1]: I think this is just naming conventions from `dsp-appsec-blessed-images`. The "-debian" image is built on top of Temurin: https://github.com/broadinstitute/dsp-appsec-blessed-images/blob/main/jre/Dockerfile.17-debian

**Pull request merged**